commerce_coordinator/apps/frontend_app_ecommerce/views.py

```diff
@@ -25,6 +30,41 @@ def date_conv(dt: Union[datetime, str]) -> datetime:
         return dt
 
 
+# noinspection PyMethodMayBeStatic
+class RedirectReceiptView(APIView):
+    """Get the order history for the authenticated user."""
+    permission_classes = [LoginRedirectIfUnauthenticated]
+    throttle_classes = [UserRateThrottle]
+
+    def get(self, request):
+        """Get the order history for the authenticated user."""
+
+        # build parameters
+        params = {
+            'username': request.user.username,
+            "edx_lms_user_id": request.user.lms_user_id,
+            "order_number": request.query_params.get('order_number', None),
+        }
+
+        # deny global queries
+        if not request.user.username:  # pragma: no cover
+            # According to the Django checks, this isn't possible with our current user model.
+            # Leaving in incase that changes.
+            raise PermissionDenied(detail="Could not detect username.")
+        if not request.user.lms_user_id:  # pragma: no cover
+            raise PermissionDenied(detail="Could not detect LMS user id.")
```

Comment:
IMO, we don't need these checks here. Our JWT authentication ensures to creation of a new user in the service and mapp its LMS ID and username.

Reply:
We were doing this in other places... So i figured following the previous pattern was more correct.